tests/src/core/testqgsmaplayer.cpp

```diff
@@ -387,5 +390,59 @@ void TestQgsMapLayer::notify()
   QCOMPARE( spyDataChanged.count(), 2 );
 }
 
+void TestQgsMapLayer::customEnumFlagProperties()
+{
+  QgsVectorLayer *ml = new QgsVectorLayer( QStringLiteral( "Point" ), QStringLiteral( "name" ), QStringLiteral( "memory" ) );
```

Comment:
```suggestion
  std::unique_ptr<QgsVectorLayer> ml = std::make_unique<QgsVectorLayer>( QStringLiteral( "Point" ), QStringLiteral( "name" ), QStringLiteral( "memory" ) );
```

Let's try to not leak in unit tests either